### PR TITLE
Insert Jobs in batch

### PR DESCRIPTION
**Pull request by GuruPKK**

### CODE REVIEW

File: api/controllers/v1/slack.ts [hunk 1]
```diff
@@ -193,110 +188,42 @@ export const DeployRepo = async (event: any = {}, context: any = {}): Promise<an
...
-        );
-        await deployRepo(
-          createJob(newPayload, jobTitle, jobUserName, jobUserEmail),
-          consoleLogger,
-          jobRepository,
-          parallelJobRepo,
-          parallelUrl,
-          parallelPrefix
-        );
+        newPayload.urlSlug = '';
```
Comment (rayangler) [inline api/controllers/v1/slack.ts:204]: This assignment may already be covered when we first initialize `newPayload`:
```
non_versioned ? '' : branchName,
```
Should we remove this additional assignment or should we keep it here for transparency?

File: src/repositories/jobRepository.ts [hunk 2]
```diff
@@ -54,10 +54,27 @@ export class JobRepository extends BaseRepository {
...
     // Insertion/re-enqueueing should be sent to jobs queue and updates for an existing job should be sent to jobUpdates Queue
+
     await this._queueConnector.sendMessage(new JobQueueMessage(jobId, JobStatus.inQueue), url, 0);
     return jobId;
   }
 
+  async insertJBulkJobs(jobs: Array<any>, url: string): Promise<Array<any>> {
+    const jobIds = await this.insertMany(jobs, `Mongo Timeout Error: Timed out while inserting bulk Jobs`);
+    if (!jobIds) {
+      throw new DBError('insertJBulkJobs:Unable to insert multiple jobs');
```
Comment (rayangler) [inline src/repositories/jobRepository.ts:65]: (nit) spacing
```suggestion
      throw new DBError('insertJBulkJobs: Unable to insert multiple jobs');
```

File: src/repositories/jobRepository.ts [hunk 3]
```diff
@@ -54,10 +54,27 @@ export class JobRepository extends BaseRepository {
...
+    }
+    // Insertion/re-enqueueing should be sent to jobs queue and updates for an existing job should be sent to jobUpdates Queue
+    console.log(`Total Jobs Expected : ${jobs.length}, Total Jobs Sent: ${jobIds.length}`);
+    await Promise.all(
+      Object.values(jobIds).map(async (jobId: string) => {
+        await this._queueConnector.sendMessage(new JobQueueMessage(jobId, JobStatus.inQueue), url, 0);
+        console.log(jobId);
```
Comment (rayangler) [inline src/repositories/jobRepository.ts:72]: Are these console logs intentional?

File: api/controllers/v1/jobs.ts [hunk 4]
```diff
@@ -76,7 +76,8 @@ export const HandleJobs = async (event: any = {}): Promise<any> => {
         }
       } catch (err) {
         consoleLogger.error(jobId, err);
-        await retry(body, consoleLogger, queueUrl);
+        throw err;
+        // await retry(body, consoleLogger, queueUrl);
```
Comment (rayangler) [inline api/controllers/v1/jobs.ts:80]: Do we need to keep this commented?

File: api/controllers/v1/jobs.ts [hunk 5]
```diff
@@ -137,32 +137,18 @@ async function NotifyBuildSummary(jobId: string): Promise<any> {
...
-    let end = 255;
-    while (message[end] != ' ') {
-      end -= 1;
-    }
-    message = message.substring(0, end + 1);
-  }
-  return message;
-}
-
 function extractUrlFromMessage(fullDocument) {
```
Comment (mango-db) [inline api/controllers/v1/jobs.ts:140]: Should we add a couple quick unit tests for this function?

File: api/controllers/v1/jobs.ts [hunk 6]
```diff
@@ -114,20 +115,19 @@ async function NotifyBuildSummary(jobId: string): Promise<any> {
   const username = fullDocument.user;
   const slackConnector = new SlackConnector(consoleLogger, c);
   const repoEntitlementRepository = new RepoEntitlementsRepository(db, c, consoleLogger);
-  const entitlement = await repoEntitlementRepository.getRepoEntitlementsByGithubUsername(username);
+  const entitlement = await repoEntitlementRepository.getSlackUserIdByGithubUsername(username);
+  console.log(entitlement, username, jobId);
   if (!entitlement?.['slack_user_id']) {
+    console.log('Entitlement failed');
```
[the remark this answers is not in the document]
Comment (GuruPKK) [reply api/controllers/v1/jobs.ts]: Removed unwanted logs and replaced it consoleLogger for debugging purposes

File: api/controllers/v1/slack.ts [hunk 7]
```diff
@@ -153,6 +155,7 @@ export const DeployRepo = async (event: any = {}, context: any = {}): Promise<an
     let aliases = branchObject.aliasObject.urlAliases; //array or null
     let urlSlug = branchObject.aliasObject.urlSlug; //string or null, string must match value in urlAliases or gitBranchName
     const isStableBranch = branchObject.aliasObject.isStableBranch; // bool or Falsey
+    console.log(repoName, branchName, branchObject.aliasObject);
```
[the remark this answers is not in the document]
Comment (GuruPKK) [reply api/controllers/v1/slack.ts]: Removed

File: api/controllers/v1/slack.ts [hunk 8]
```diff
@@ -193,110 +188,42 @@ export const DeployRepo = async (event: any = {}, context: any = {}): Promise<an
...
-        await deployRepo(
-          createJob(newPayload, jobTitle, jobUserName, jobUserEmail),
-          consoleLogger,
-          jobRepository,
-          parallelJobRepo,
-          parallelUrl,
-          parallelPrefix
-        );
+        newPayload.urlSlug = urlSlug;
+        console.log('Present URLSLUG', urlSlug, newPayload);
```
[the remark this answers is not in the document]
Comment (GuruPKK) [reply api/controllers/v1/slack.ts]: Removed

File: api/controllers/v1/slack.ts (hunk 1 above)
Comment (GuruPKK) [reply api/controllers/v1/slack.ts]: removed

File: src/repositories/jobRepository.ts (hunk 3 above)
Comment (GuruPKK) [reply src/repositories/jobRepository.ts]: Removed

File: api/controllers/v1/jobs.ts (hunk 5 above)
Comment (GuruPKK) [reply api/controllers/v1/jobs.ts:139]: sure

File: api/controllers/v1/jobs.ts (hunk 4 above)
Comment (GuruPKK) [reply api/controllers/v1/jobs.ts:80]: Removed

**Review by rayangler — APPROVED** [review]
LGTM

File: api/controllers/v1/jobs.ts (hunk 5 above)
Comment (GuruPKK) [reply api/controllers/v1/jobs.ts]: added

**Pull request merged**